### ISSUE TITLE
Python API: close cursor at the end of the transaction

### ISSUE THREAD

**Issue by edigiacomo** [ARPA-SIMC/dballe#171]
Now, to ensure that the cursor will be closed and freed we must write:

```
with db.transaction() as tr:
    with tr.query_data() as cursor:
        for row in cursor:
            ...
```
We have a lot of scripts with the following template:

```
with db.transaction() as tr:
    for row in query_data() as cursor:
        ...
```
I don't know if the cursor is deallocated by the GC. In these cases it would be nice and useful to automatically close the cursor at the end of the transaction, without the with statement for the cursor.

Maybe the explicit with statement for the cursor can still be useful when there are a lot of cursors within the same transaction.

```
with db.transaction() as tr:
    for i in range(LARGE_NUMBER):
        with tr.query_data({...}) as cur:
             for row in cur:
                 ...
```

**Comment by edigiacomo** [comment]
Per me benissimo!

**Comment by spanezz** [comment]
Fatto push nel branch issue171. Aspetto l'OK prima di far merge su master, perché col deploy di questo dballe serve ricompilare i programmi che lo usano.

**Comment by dcesari** [comment]
Come detto per altre vie, wreport e bufr2netcdf hanno pochi ma utili aggiornamenti committed in master ma non released, stesso discorso per libsim che ha l'aggiornamento del formato json, per cui ben venga una release vera per loro.

**Comment by pat1** [comment]
io sono in attesa di questa, ma direi essere indipendente:
https://github.com/ARPA-SIMC/dballe/issues/235

**Comment by brancomat** [comment]
Now released in dballe v9.0
